packages/app/src/systems/Transaction/machines/transactionMachine.tsx

```diff
@@ -0,0 +1,223 @@
+  };
+};
+
+type MachineEvents = {
+  type: 'GET_TRANSACTION';
+  input: { providerUrl?: string; txId?: string };
+};
+
+export const transactionMachine = createMachine(
+  {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import('./transactionMachine.typegen').Typegen0,
+    schema: {
+      context: {} as MachineContext,
+      services: {} as MachineServices,
+      events: {} as MachineEvents,
+    },
+    predictableActionArguments: true,
+    id: '(machine)',
+    initial: 'idle',
+    states: {
+      idle: {
+        on: {
+          GET_TRANSACTION: [
+            {
+              cond: 'isInvalidTxId',
+              actions: 'assignInvalidTxIdError',
+            },
+            {
+              target: 'fetching',
+            },
+          ],
+        },
+      },
+      fetching: {
+        entry: 'clearError',
+        invoke: {
+          src: 'getTransaction',
+          data: (_, event: MachineEvents) => ({
+            input: event.input,
+          }),
+          onDone: [
+            {
+              actions: ['assignGetTransactionError'],
+              target: 'idle',
+              cond: FetchMachine.hasError,
+            },
+            {
+              actions: ['assignGetTransactionResponse'],
+              target: 'fetchingResult',
+            },
+          ],
+        },
+      },
+      fetchingResult: {
+        invoke: {
+          src: 'getTransactionResult',
+          data: (ctx) => ({
+            input: {
+              txResponse: ctx.txResponse,
+            },
+          }),
+          onDone: [
+            {
+              actions: ['assignGetTransactionResultError'],
+              target: 'idle',
+              cond: FetchMachine.hasError,
+            },
+            {
+              actions: ['assignGetTransactionResult'],
+              target: 'idle',
+            },
+          ],
+        },
+      },
+    },
+  },
+  {
+    actions: {
+      assignInvalidTxIdError: assign({
+        error: (_) => TRANSACTION_ERRORS.INVALID_ID,
+      }),
+      assignGetTransactionError: assign({
+        error: (_) => TRANSACTION_ERRORS.NOT_FOUND,
+      }),
+      assignGetTransactionResultError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
+      clearError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
+      assignGetTransactionResponse: assign((_, event) => {
+        const data = event.data as GetTransactionResponse;
+
+        return {
+          txResponse: data.txResponse,
+          tx: data.tx,
+          txStatus: data.status,
+          txId: data.txId,
+        };
+      }),
+      assignGetTransactionResult: assign((_, event) => {
+        const txResult = event.data as TransactionResult<any>;
+
+        return {
+          txResult,
+          transaction: txResult.transaction,
+          txStatus:
+            txResult.status.type === 'success'
```

Comment:
Do we have an enum for status type? I think `fuels` has.

Reply:
it doesn't:

<img width="725" alt="image" src="https://user-images.githubusercontent.com/8636507/206508436-5d478719-90d8-4f63-83ba-d710174a34ea.png">